cfme/tests/infrastructure/test_providers.py

```diff
@@ -514,14 +514,13 @@ def test_compare_provider_templates(appliance, setup_provider_min_templates, pro
     Bugzilla:
         1746449
     """
-    templateList = []
-    my_slice = slice(0, min_templates, None)
-    view = navigate_to(provider, 'ProviderTemplates')
-    for t in view.entities.get_all(slice=my_slice):
-        t.ensure_checked()
-        templateList.append(t.name)
-    view.toolbar.configuration.item_select('Compare Selected Templates', handle_alert=True)
-    compare_templates_view = provider.create_view(TemplatesCompareView)
-    assert compare_templates_view.is_displayed
-    assert compare_templates_view.verify_checked_items_compared(templateList,
-                                                                compare_templates_view)
+    t_coll = appliance.collections.infra_templates.all()[:min_templates]
+    compare_view = locals()[templates_collection].collections.infra_templates.compare_entities(
+        provider,
+        entities_list=t_coll)
+    assert compare_view.is_displayed
+
+    t_list = []
```

Comment:
Optional, consider using list comprehension here. `templates = [t.name for t in t_coll]`

Reply:
Of course.

Reply:
FYI. I still have a rewrite of verify_checked_items_compared on my todo list.